src/test/java/org/broadinstitute/hellbender/tools/walkers/mutect/DuplicateReadCountsUnitTest.java

```diff
@@ -0,0 +1,125 @@
+        uniqueAltReadCountAnnotation.annotate(null, vc, genotypeBuilder.make(), genotypeBuilder, likelihoods);
+
+        final Genotype genotype = genotypeBuilder.make();
+
+        final int uniqueReadSetCount = GATKProtectedVariantContextUtils.getAttributeAsInt(genotype, UniqueAltReadCount.UNIQUE_ALT_READ_SET_COUNT_KEY, -1);
+        Assert.assertEquals(uniqueReadSetCount, 1);
+    }
+
+    @Test
+    public void testMultipleDuplicateSets() throws IOException {
+        final UniqueAltReadCount duplicateReadCountsAnnotation = new UniqueAltReadCount();
+
+        // should get three unique sets of ALT reads
+        final int shiftBy1 = 3;
+        final ReadLikelihoods<Allele> likelihoods1 = createTestLikelihoods(true, Optional.of(shiftBy1));
+        final GenotypeBuilder genotypeBuilder1 = new GenotypeBuilder(sampleName);
+        duplicateReadCountsAnnotation.annotate(null, vc, genotypeBuilder1.make(), genotypeBuilder1, likelihoods1);
+
+        final Genotype genotype1 = genotypeBuilder1.make();
+
+        final int uniqueReadSetCount1 = GATKProtectedVariantContextUtils.getAttributeAsInt(genotype1, UniqueAltReadCount.UNIQUE_ALT_READ_SET_COUNT_KEY, -1);
+
+        Assert.assertEquals(uniqueReadSetCount1, shiftBy1);
+
+        // here ALT reads are all distinct
+        final int shiftBy2 = 12;
+        final ReadLikelihoods<Allele> likelihoods2 = createTestLikelihoods(true, Optional.of(shiftBy2));
+        final GenotypeBuilder genotypeBuilder2 = new GenotypeBuilder(sampleName);
+        duplicateReadCountsAnnotation.annotate(null, vc, genotypeBuilder2.make(), genotypeBuilder2, likelihoods2);
+
+        final Genotype genotype2 = genotypeBuilder2.make();
+
+        final int uniqueReadSetCount2 = GATKProtectedVariantContextUtils.getAttributeAsInt(genotype2, UniqueAltReadCount.UNIQUE_ALT_READ_SET_COUNT_KEY, -1);
+
+        Assert.assertEquals(uniqueReadSetCount2, shiftBy2);
+    }
+
+
+    private ReadLikelihoods<Allele> createTestLikelihoods(final boolean shiftReads, Optional<Integer> shiftBy) {
+        final int numChromosomes = 2;
+        final int startingChromosome = 1;
+        final int chromosomeSize = 1000;
+        final SAMFileHeader SAM_HEADER = ArtificialReadUtils.createArtificialSamHeader(numChromosomes, startingChromosome, chromosomeSize);
+
+        final int alignmentStart = 100;
+        final int readLength = 11;
+        final byte baseq = 30;
+
+        final int numAltReads = 12;
+        final int numRefReads = 10;
+        final int numReads = numAltReads + numRefReads;
+
+        final Map<String, List<GATKRead>> readMap = new LinkedHashMap<>();
+        final List<GATKRead> reads = new ArrayList<>(numReads);
+        final byte[] quals = new byte[readLength];
+        Arrays.fill(quals, baseq);
+
+        for (int i = 0; i < numAltReads; i++) {
+            final int startPosition = shiftReads ? alignmentStart + (i % shiftBy.get()) : alignmentStart;
```

Comment:
Could you change the name of shiftBy to something like, nUniqueStartStopPositions?
Also, could you add a comment here explaining how you are creating different unique start stop positions by shifting using the mod?